endpoint.go

```diff
@@ -226,14 +215,10 @@ func (c cachePurge) handlePurge(w http.ResponseWriter, r *http.Request) error {
 
 	purgeRepl.Set("http.request.method", payload.Method)
 	purgeRepl.Set("http.request.host", payload.Host)
-	purgeRepl.Set("http.request.uri.query", payload.query)
-	purgeRepl.Set("http.request.uri.path", payload.path)
+	purgeRepl.Set("http.request.uri", payload.URI)
 
 	cache := getHandlerCache()
-	// example key should be like "GET localhost/static/js/chunk-element.js?"
-	i := strings.Index(config.CacheKeyTemplate, "?")
-	escapedKeyTmpl := config.CacheKeyTemplate[:i] + "\\" + config.CacheKeyTemplate[i:]
 
-	conds := purgeRepl.ReplaceKnown(escapedKeyTmpl, "")
-	return c.Purge(cache, conds)
+	conds := purgeRepl.ReplaceKnown(config.CacheKeyTemplate, "")
+	return c.Purge(cache, regexp.QuoteMeta(conds))
```

Comment:
You should not use the `regexp.QuoteMeaa` directly because I just want the `?` to be used as a separate of URL path and URL query not for the regular expression.

If you use `regexp.QuoteMeta here the whole conds will be escaped, that will not work as I expect.

ex.
If I want to purge all possible cache keys with a URL query with prefix `a` then the conds will look like
`GET example.com/path\?a.*`